File: server/natsuki-framework/src/main/java/top/natsuki/domain/entity/Article.java
package top.natsuki.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * 
 * @TableName na_article
 */
@TableName(value ="na_article")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain=true)
public class Article
//        implements Serializable
{
    /**
     * 文章ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 文章标题
     */
    private String title;

    /**
     * 文章内容
     */
    private String content;

    /**
     * 文章摘要
     */
    private String summary;

    /**
     * 文章分类ID
     */
    private Long categoryId;

    /**
     * 分类名称
     */
    @TableField(exist = false)
    private String categoryName;

    /**
     * 文章缩略图
     */
    private String thumbnail;

    /**
     * 是否置顶（0否，1是）
     */
    private Integer isTop;

    /**
     * 文章状态（0已发布，1代表草稿）
     */
    private Integer status;

    /**
     * 浏览量
     */
    private Long viewCount;

    /**
     * 是否允许评论（0否，1是）
     */
    private Integer isComment;

    /**
     * 创建者
     */
    private Long createBy;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新者
     */
    private Long updateBy;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

    /**
     * 删除标志（ 0 代表存在 1 代表删除）
     */
    private Integer delFlag;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}

File: client/src/types/article.ts
// src/types/article.ts

// 1. 定义单篇文章的结构
// 根据你提供的 JSON，把所有字段都定义出来，不确定的可以用 null 联合类型
export interface Article {
    id: number;
    title: string;
    // summary: string | null;
    // content: string;
    viewCount: number;      // 热门文章通常看这个
    // isTop: number;          // 1 代表置顶，可以用作特殊的 UI 标记
    // thumbnail: string | null;
    // createTime: string;
    // categoryId: number;
    // ... 其他字段根据需要添加，暂时这些够用了
}

// 2. 定义通用的 API 响应结构
// 这是一个泛型接口 (Generic Interface)，<T> 代表 data 的类型是可变的
export interface ApiResponse<T> {
    code: number;
    msg: string;
    data: T;
}